Fix:

error: incompatible function pointer types assigning to 'void (*)(PangoEngineShape *, PangoFont *, const char *, unsigned int, const PangoAnalysis *, PangoGlyphString *, const char *, unsigned int)' (aka 'void (*)(struct _PangoEngineShape *, struct _PangoFont *, const char *, unsigned int, const struct _PangoAnalysis *, struct _PangoGlyphString *, const char *, unsigned int)') from 'void (PangoEngineShape *, PangoFont *, const char *, gint, const PangoAnalysis *, PangoGlyphString *)' (aka 'void (struct _PangoEngineShape *, struct _PangoFont *, const char *, int, const struct _PangoAnalysis *, struct _PangoGlyphString *)') [-Wincompatible-function-pointer-types]

https://bugs.freebsd.org/bugzilla/show_bug.cgi?id=271709
https://cgit.freebsd.org/ports/commit/?id=1f905384aff97cb61995ae241e77261bc109a865
--- basic-x.c.orig	2012-08-27 23:19:32 UTC
+++ basic-x.c
@@ -556,9 +556,11 @@ basic_engine_shape (PangoEngineShape *engine G_GNUC_UN
 basic_engine_shape (PangoEngineShape *engine G_GNUC_UNUSED,
 		    PangoFont        *font,
 		    const char       *text,
-		    gint              length,
+		    unsigned int      length,
 		    const PangoAnalysis *analysis,
-		    PangoGlyphString *glyphs)
+		    PangoGlyphString *glyphs,
+		    const char       *paragraph_text G_GNUC_UNUSED,
+		    unsigned int      paragraph_length G_GNUC_UNUSED)
 {
   int n_chars;
   int i;
